vendor/github.com/livepeer/lpms/stream/basic_rtmp_videostream.go

```diff
@@ -37,97 +63,57 @@ func (s *BasicRTMPVideoStream) GetStreamFormat() VideoFormat {
 func (s *BasicRTMPVideoStream) ReadRTMPFromStream(ctx context.Context, dst av.MuxCloser) error {
 	defer dst.Close()
 
-	//TODO: Make sure to listen to ctx.Done()
-	for {
-		item, err := s.buffer.poll(ctx, s.RTMPTimeout)
-		if err != nil {
-			return err
-		}
+	if err := dst.WriteHeader(s.header); err != nil {
+		return err
+	}
 
-		switch item.(type) {
-		case []av.CodecData:
-			headers := item.([]av.CodecData)
-			err = dst.WriteHeader(headers)
-			if err != nil {
-				glog.Errorf("Error writing RTMP header from Stream %v to mux", s.streamID)
-				return err
-			}
-		case av.Packet:
-			packet := item.(av.Packet)
-			err = dst.WritePacket(packet)
-			if err != nil {
-				glog.Errorf("Error writing RTMP packet from Stream %v to mux: %v", s.streamID, err)
-				return err
-			}
-		case RTMPEOF:
-			err := dst.WriteTrailer()
-			if err != nil {
-				glog.Errorf("Error writing RTMP trailer from Stream %v", s.streamID)
-				return err
-			}
-			return io.EOF
-		default:
-			glog.Errorf("Cannot recognize buffer iteam type: ", reflect.TypeOf(item))
-			debug.PrintStack()
-			return ErrBufferItemType
+	s.listnersLock.Lock()
+	s.listeners = append(s.listeners, dst)
+	s.listnersLock.Unlock()
+
+	select {
```

Comment:
Is this select statement supposed to just block until we reach the end of the RTMP stream or the context sends a done signal?

Reply:
Yeah - we basically wait until s.EOF gets closed, which happens when the writing loop gets io.EOF. (This also kills the worker loop created in `NewBasicRTMPVideoStream`)